cl/search/api_utils.py

```diff
@@ -247,12 +286,211 @@ def __getitem__(self, item):
+        self.process_results(self.results)
+        return (
+            self.results,
+            cardinality_count_response,
+            child_cardinality_count_response,
+        )
+
+    def process_results(self, results: Response) -> None:
+        """Processes the raw results from ES for handling inner hits,
+        highlighting and merging of unavailable fields.
+        """
+
+        limit_inner_hits({}, results, self.clean_data["type"])
+        set_results_highlights(results, self.clean_data["type"])
+        merge_unavailable_fields_on_parent_document(
+            results,
+            self.clean_data["type"],
+            "api",
+            self.clean_data["highlight"],
+        )
+        for result in results:
+            child_result_objects = []
+            if hasattr(result, "child_docs"):
+                for child_doc in result.child_docs:
+                    child_result_objects.append(
+                        ESResultObject(initial=child_doc["_source"])
+                    )
+                result["child_docs"] = child_result_objects
+
+        if self.reverse:
+            # If doing backward pagination, reverse the results of the current
+            # page to maintain consistency of the results on the page,
+            # because the original order is inverse when paginating backwards.
+            self.results.hits.reverse()
+
+    def _get_search_after_key(self, index_position: int) -> AttrList | None:
+        if self.results and len(self.results) > 0:
+            limited_results = limit_api_results_to_page(
+                self.results.hits, self.cursor
+            )
+            last_result = limited_results[index_position]
+            return last_result.meta.sort
+        return None
+
+    def get_search_after_sort_key(self) -> AttrList | None:
+        """Retrieves the sort key from the last item in the current page to
+        use for the next page's search_after parameter.
+        """
+        last_result_in_page = -1
+        return self._get_search_after_key(last_result_in_page)
+
+    def get_reverse_search_after_sort_key(self) -> AttrList | None:
+        """Retrieves the sort key from the last item in the current page to
+        use for the next page's search_after parameter.
+        """
+        first_result_in_page = 0
+        return self._get_search_after_key(first_result_in_page)
+
+    def get_api_query_sorting(self):
+        """Build the sorting settings for an ES query to work with the
+        'search_after' pagination. Two sorting keys are returned: the default
+        sorting requested by the user and a unique sorting key based on a
+        unique field across documents, acting as a tiebreaker for the default
+        sorting.
+
+        :return: A tuple containing default_sorting and unique_sorting
+        settings.
+        """
+
+        # Toggle the original sorting key to handle backward pagination
+        toggle_sort = True if self.reverse else False
```

Comment:
Let's set a default value of `False` for the `reverse` variable. This should simplify the code like this: 

```python
toggle_sort = self.reverse
```

We can also avoid the need to explicitly assign it in most cases.

Reply:
Sure, doing this we can even use `self.reverse` directly without requiring `toggle_sort`, changed it.